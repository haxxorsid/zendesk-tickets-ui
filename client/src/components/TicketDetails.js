import React from 'react';
import Modal from 'react-modal';
import { formattedDate } from '../common/helper';

function TicketDetails (props) {
    return (
        <Modal
          isOpen={!!props.selectedTicket}
          contentLabel="Details"
          onRequestClose={props.onClearSelectedTicket}
          className="modal"
          ariaHideApp={false}
        >
          {props.selectedTicket && 
            <div>
              <h4>{props.selectedTicket[0].subject}</h4>
              <div className="ticket-header">
                <p>
                  <span className="status"> Status: {props.selectedTicket[0].status}</span>
                  <span className="right date ">{formattedDate(props.selectedTicket[0].created_at)}</span> 
                </p>
              </div>
              <p className="ticket-details-description">{props.selectedTicket[0].description}</p> 
            </div>
          }
          <button className="button" onClick={props.onSelectedTicketClear}>Back</button>
        </Modal>
    );
}

export default TicketDetails;