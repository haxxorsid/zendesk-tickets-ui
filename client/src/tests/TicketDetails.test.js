import React from 'react';
import renderer from 'react-test-renderer'
import TicketDetails from '../components/TicketDetails';
import ticketData from './mockData/ticket';
import ReactDOM from 'react-dom';

test('Should render ticket details correctly with props', () => {
    const onSelectedTicketClearFn = jest.fn();
    ReactDOM.createPortal = node => node;
    const component = renderer
        .create(<TicketDetails selectedTicket={[ticketData]} onClearSelectedTicket={onSelectedTicketClearFn} />);
    expect(component.toJSON()).toMatchSnapshot()
})