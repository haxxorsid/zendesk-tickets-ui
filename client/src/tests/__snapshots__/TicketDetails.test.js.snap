// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ticket details correctly with props 1`] = `
<div
  className="ReactModal__Overlay"
  onClick={[Function]}
  onMouseDown={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 0.75)",
      "bottom": 0,
      "left": 0,
      "position": "fixed",
      "right": 0,
      "top": 0,
    }
  }
>
  <div
    aria-label="Details"
    aria-modal={true}
    className="ReactModal__Content modal"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    role="dialog"
    style={Object {}}
    tabIndex="-1"
  >
    <div>
      <h4>
        velit eiusmod reprehenderit officia cupidatat
      </h4>
      <div
        className="ticket-header"
      >
        <p>
          <span
            className="status"
          >
             Status: 
            open
          </span>
          <span
            className="right date "
          >
            Tue Nov 23 2021
          </span>
        </p>
      </div>
      <p
        className="ticket-details-description"
      >
        Aute ex sunt culpa ex ea esse sint cupidatat aliqua ex consequat sit reprehenderit. Velit labore proident quis culpa ad duis adipisicing laboris voluptate velit incididunt minim consequat nulla. Laboris adipisicing reprehenderit minim tempor officia ullamco occaecat ut laborum.

Aliquip velit adipisicing exercitation irure aliqua qui. Commodo eu laborum cillum nostrud eu. Mollit duis qui non ea deserunt est est et officia ut excepteur Lorem pariatur deserunt.
      </p>
    </div>
    <button
      className="button"
    >
      Back
    </button>
  </div>
</div>
`;
